modules/extension/xsd/xsd-fes/src/test/java/org/geotools/filter/v2_0/bindings/BBoxBindingTest.java

```diff
@@ -0,0 +1,98 @@
+package org.geotools.filter.v2_0.bindings;
+
+import java.io.IOException;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.filter.v2_0.FES;
+import org.geotools.filter.v2_0.FESTestSupport;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.referencing.CRS;
+import org.geotools.xsd.Encoder;
+import org.junit.Assert;
+import org.junit.Test;
+import org.opengis.filter.Filter;
+import org.opengis.filter.FilterFactory2;
+import org.opengis.filter.expression.Literal;
+import org.opengis.filter.expression.PropertyName;
+import org.opengis.filter.spatial.BBOX;
+import org.opengis.geometry.MismatchedDimensionException;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.NoSuchAuthorityCodeException;
+
+public class BBoxBindingTest extends FESTestSupport {
+
+    @Test
+    public void testEncode() {
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+
+        try {
+            ReferencedEnvelope env =
+                    new ReferencedEnvelope(
+                            Double.valueOf(48.81752162),
+                            Double.valueOf(48.81915540),
+                            Double.valueOf(-3.47568429),
+                            Double.valueOf(-3.47261370),
+                            CRS.decode("EPSG:4326"));
+            Filter bboxFilter = ff.bbox(ff.property("geom"), env);
+            Encoder encoder = new Encoder(createConfiguration());
+            String encodedFilter = encoder.encodeAsString(bboxFilter, FES.Filter);
+            Assert.assertTrue(
+                    encodedFilter.contains("<fes:ValueReference>geom</fes:ValueReference>"));
+            Assert.assertTrue(
+                    encodedFilter.contains(
+                            "<gml:Envelope srsDimension=\"2\" srsName=\"urn:ogc:def:crs:EPSG::4326\">"));
+            Assert.assertTrue(
+                    encodedFilter.contains(
+                            "<gml:lowerCorner>48.817522 -3.475684</gml:lowerCorner>"));
+            Assert.assertTrue(
+                    encodedFilter.contains(
+                            "<gml:upperCorner>48.819155 -3.472614</gml:upperCorner>"));
+
+        } catch (MismatchedDimensionException e) {
```

Comment:
Letting it simply throw the exception is an easier way to get the test to fail, also gives all details including a stack trace, while this only reports the error message.

Reply:
OK, Unit test catch exception removed